src/navigators/createMaterialBottomTabNavigator.js

```diff
@@ -36,28 +29,31 @@ class BottomNavigationView extends React.Component {
       descriptors,
       barStyle,
       theme,
+      sendEventForTab,
       ...rest
     } = this.props;
-
     const isVisible = this._isVisible();
     const extraStyle =
-      typeof isVisible === 'boolean'
-        ? { display: isVisible ? null : 'none' }
+      typeof isVisible === "boolean"
+        ? { display: isVisible ? null : "none" }
         : null;
-
     return (
       <BottomNavigation
         // Pass these for backward compaibility
         {...rest}
         activeColor={theme.colors.activeColor}
-        inactiveColor={theme.colors.inactiveColor} 
+        inactiveColor={theme.colors.inactiveColor}
         renderIcon={this._renderIcon}
-        barStyle={[barStyle, extraStyle,{backgroundColor: theme.colors.bg}]}
+        barStyle={[barStyle, extraStyle, { backgroundColor: theme.colors.bg }]}
         navigationState={navigation.state}
         getColor={this._getColor}
+        onIndexChange={(index) => {
+          if (sendEventForTab) {
+            sendEventForTab(navigation.state, index);
```

Comment:
Can we check and call onIndexChange from props if present? @abhishek-bharadwaj 

`onIndexChange={(index) => {
          if (sendEventForTab) {
            sendEventForTab(navigation.state, index);
          }
          if(this.props.onIndexChange){
            this.props.onIndexChange(index)
          }
}}`